typescript/src/lib/bitcoin/tx.ts

```diff
@@ -0,0 +1,237 @@
+import { Transaction as Tx } from "bitcoinjs-lib"
+import bufio from "bufio"
+import { BigNumber } from "ethers"
+import { Hex } from "../utils"
+
+/**
+ * Represents a Bitcoin transaction hash (or transaction ID) as an un-prefixed hex
+ * string. This hash is supposed to have the same byte order as used by the
+ * Bitcoin block explorers which is the opposite of the byte order used
+ * by the Bitcoin protocol internally. That means the hash must be reversed in
+ * the use cases that expect the Bitcoin internal byte order.
+ */
+export class BitcoinTxHash extends Hex {}
+
+/**
+ * Represents a raw Bitcoin transaction.
+ */
+export interface BitcoinRawTx {
+  /**
+   * The full transaction payload as an un-prefixed hex string.
+   */
+  transactionHex: string
+}
+
+/**
+ * Data about a Bitcoin transaction.
+ */
+export interface BitcoinTx {
+  /**
+   * The transaction hash (or transaction ID) as an un-prefixed hex string.
+   */
+  transactionHash: BitcoinTxHash
+
+  /**
+   * The vector of transaction inputs.
+   */
+  inputs: BitcoinTxInput[]
+
+  /**
+   * The vector of transaction outputs.
+   */
+  outputs: BitcoinTxOutput[]
+}
+
+/**
+ * Data about a Bitcoin transaction outpoint.
+ */
+export interface BitcoinTxOutpoint {
+  /**
+   * The hash of the transaction the outpoint belongs to.
+   */
+  transactionHash: BitcoinTxHash
+
+  /**
+   * The zero-based index of the output from the specified transaction.
+   */
+  outputIndex: number
+}
+
+/**
+ * Data about a Bitcoin transaction input.
+ */
+export type BitcoinTxInput = BitcoinTxOutpoint & {
+  /**
+   * The scriptSig that unlocks the specified outpoint for spending.
+   */
+  scriptSig: Hex
+}
+
+/**
+ * Data about a Bitcoin transaction output.
+ */
+export interface BitcoinTxOutput {
+  /**
+   * The 0-based index of the output.
+   */
+  outputIndex: number
+
+  /**
+   * The value of the output in satoshis.
+   */
+  value: BigNumber
+
+  /**
+   * The receiving scriptPubKey.
+   */
+  scriptPubKey: Hex
+}
+
+/**
+ * Data about a Bitcoin unspent transaction output.
+ */
+export type BitcoinUtxo = BitcoinTxOutpoint & {
+  /**
+   * The unspent value in satoshis.
+   */
+  value: BigNumber
+}
+
+/**
+ * Represents a raw Bitcoin transaction decomposed into specific vectors.
+ */
+export interface BitcoinRawTxVectors {
+  /**
+   * Transaction version as an un-prefixed hex string.
+   */
+  version: string
```

Comment:
We're not consistent with using `Hex` and `string` types around the interfaces.

Reply:
Yeah, and https://github.com/keep-network/tbtc-v2/issues/713 will solve this.